src/main/java/net/zenoc/gallium/plugin/PluginManager.java

```diff
@@ -60,52 +62,97 @@ public void loadPlugins() throws IOException {
             log.info("Loading plugin {}", file.getName());
             ZipFile zip = new ZipFile(file);
             if (zip.getEntry("Canary.inf") != null) {
-                log.warn("CanaryMod plugins are not natively supported. Please remove {}", file.getName());
+                throw new BadPluginException("CanaryMod plugins are not natively supported. Please remove " + file.getName());
             } else if (zip.getEntry("plugin.yml") != null) {
-                log.warn("Bukkit plugins are not natively supported. Please remove {}", file.getName());
+                throw new BadPluginException("Bukkit plugins are not natively supported. Please remove " + file.getName());
             } else if (zip.getEntry("bungee.yml") != null) {
-                log.warn("BungeeCord plugins are not natively supported. Please remove {}", file.getName());
+                throw new BadPluginException("BungeeCord plugins are not natively supported. Please remove " + file.getName());
             } else if (zip.getEntry("mcmod.info") != null) {
-                log.warn("Sponge plugins and Forge mods aren ot natively supported. Please remove {}", file.getName());
+                throw new BadPluginException("Sponge plugins and Forge mods are not natively supported. Please remove " + file.getName());
```

Comment:
Seems like a weird thing to do tbh

Reply:
what?

Reply:
I don't know if it's necessary to have specific errors for other loaders/apis...

Reply:
I suppose so, but I don't see why not

Reply:
There are a lot!